experiments/dice_experiment/run_dice_experiment.py

```diff
@@ -0,0 +1,553 @@
+
+    Returns:
+        A merged set of results containing data from all_results and
+        run_results."""
+    if not all_results:
+        return run_results
+    else:
+        if set(all_results.keys()) != set(run_results.keys()):
+            raise RuntimeError(
+                (
+                    "The new results dictionary contains keys "
+                    f"{run_results.keys()}, but the original results have keys"
+                    f" {all_results.keys()}"
+                )
+            )
+
+        merged_results: Mapping[str, pd.DataFrame] = {}
+        for result_name in run_results.keys():
+            original_result = all_results[result_name]
+            new_result = run_results[result_name]
+            merged_results[result_name] = pd.concat(
+                [original_result, new_result]
+            ).reset_index(drop=True)
+
+        return merged_results
+
+
+# TODO(@jakeval): Unit test this eventually.
+def save_results(
+    results: Mapping[str, pd.DataFrame],
+    results_directory: Optional[str],
+    config: Mapping[str, Any],
+    only_csv: bool = False,
+) -> str:
+    """Saves the results and experiment config to the local file system.
+
+    Args:
+        results: A mapping from filename to DataFrame.
+        results_directory: Where to save the results.
+        config: The config used to run this experiment.
+        only_csv: Whether to exclude saving the config.json file.
+
+    Returns:
+        The directory where the results are saved."""
+    if not results_directory:
+        results_directory = os.path.join(
+            _RESULTS_DIR, config["experiment_name"]
+        )
+    if os.path.exists(results_directory):
+        shutil.rmtree(results_directory)
+    os.makedirs(results_directory)
+    for result_name, result_df in results.items():
+        result_df.to_csv(
+            os.path.join(results_directory, result_name + ".csv"), index=False
+        )
+    if not only_csv:
+        with open(
+            os.path.join(results_directory, "config.json"), "w"
+        ) as config_file:
+            json.dump(config, config_file)
+    return results_directory
+
+
+def _get_results_dir(results_directory, experiment_name):
+    return results_directory or os.path.join(_RESULTS_DIR, experiment_name)
+
+
+def run_batch(
+    run_configs: Sequence[Mapping[str, Any]],
+    verbose: bool,
+) -> Mapping[str, pd.DataFrame]:
+    """Executes a batch of runs. Each run is parameterized by a run_config.
+    The results of each run are concatenated together in DataFrames and
+    returned."""
+    all_results = None
+    for i, run_config in enumerate(run_configs):
+        paths = run_dice(**run_config)
+        run_results = format_results(paths, run_config)
+        all_results = merge_results(all_results, run_results)
+        if verbose:
+            print(f"Finished run {i+1}/{len(run_configs)}")
+    return all_results
+
+
+def validate_experiment_config(config: Mapping[str, Any]) -> None:
```

Comment:
It'd be good to combine the validation of the experiment and batch configs. This is lightweight enough that it could be included as standalone code (not abstracted into a separate function) in the `main` method.

Reply:
I combined these into the get_run_configs() function. I didn't want to add them to `main` since it's already pretty chunky.